docs/source/advanced/sagemaker-pipelines.rst

```diff
@@ -0,0 +1,303 @@
+
+5. **Training** (``train``):
+   Trains your graph neural network model.
+
+6. **Inference** (``inference``):
+   Runs predictions using your trained model.
+
+Configuration Options
+---------------------
+
+This section provides a comprehensive list of all available configuration options for creating and executing GraphStorm SageMaker pipelines.
+
+AWS Configuration
+^^^^^^^^^^^^^^^^^
+
+* ``--execution-role``: SageMaker execution IAM role ARN. (Required)
+* ``--region``: AWS region. (Required)
+* ``--graphstorm-pytorch-cpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference CPU ECR image URI. (Required)
+* ``--graphstorm-pytorch-gpu-image-uri``: GraphStorm GConstruct/dist_part/train/inference GPU ECR image URI.
+* ``--gsprocessing-pyspark-image-uri``: GSProcessing SageMaker PySpark ECR image URI. (Required if running a ``gsprocessing`` job.)
+
+Instance Configuration
+^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--instance-count`` / ``--num-parts``: Number of worker instances/partitions for partition, training, inference. (Required)
+* ``--cpu-instance-type``: CPU instance type. (Default: ml.m5.4xlarge)
+* ``--gpu-instance-type``: GPU instance type. (Default: ml.g5.4xlarge)
+* ``--train-on-cpu``: Run training and inference on CPU instances instead of GPU. (Flag)
+* ``--graph-construction-instance-type``: Instance type for graph construction.
+* ``--gsprocessing-instance-count``: Number of GSProcessing instances (PySpark cluster size, default is equal to ``--instance-count``).
+* ``--volume-size-gb``: Additional volume size for SageMaker instances in GB. (Default: 100)
+
+Task Configuration
+^^^^^^^^^^^^^^^^^^
+
+* ``--graph-name``: Name of the graph. (Required)
+* ``--input-data-s3``: S3 path to the input graph data. (Required)
+* ``--output-prefix-s3``: S3 prefix for the output data. (Required)
+* ``--pipeline-name``: Name for the pipeline.
+* ``--base-job-name``: Base job name for SageMaker jobs. (Default: 'gs')
+* ``--jobs-to-run``: Space-separated string of jobs to run in the pipeline.
+  Possible values are: ``gconstruct``, ``gsprocessing``, ``dist_part``, ``gb_convert``, ``train``, ``inference`` (Required).
+* ``--log-level``: Logging level for the jobs. (Default: INFO)
+* ``--step-cache-expiration``: Expiration time for the step cache. (Default: 30d)
+* ``--update-pipeline``: Update an existing pipeline instead of creating a new one. (Flag)
+
+Graph Construction Configuration
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--graph-construction-config-filename``: Filename for the graph construction config.
+* ``--graph-construction-args``: Additional parameters to be passed directly to the GConstruct/GSProcessing job.
+
+Partition Configuration
+^^^^^^^^^^^^^^^^^^^^^^^
+
+* ``--partition-algorithm``: Partitioning algorithm to use. (Default: random)
+* ``--partition-input-json``: Name for the JSON file that describes the input data for distributed partitioning. (Default: updated_row_counts_metadata.json)
```

Comment:
Are `--partition-input-json` and `--partition-output-json` optional?

Reply:
Yes, I think we derive them from the job sequence provided.